MapboxVision/VisionManager/VisionManager.swift

```diff
@@ -227,16 +227,16 @@ public final class VisionManager: BaseVisionManager {
         resume()
     }
 
-    private func tryRecording(mode: SessionRecorder.Mode) {
-        guard mode.isExternal || currentCountry.allowsRecording else { return }
+    private func tryRecording(mode: SessionRecordingMode) {
+        guard mode != .internal || currentCountry.allowsRecording else { return }
         dependencies.recorder.start(mode: mode)
     }
 
     override public func onCountryUpdated(_ country: Country) {
         super.onCountryUpdated(country)
         if country.allowsRecording {
             tryRecording(mode: .internal)
-        } else if dependencies.recorder.currentMode.isInternal {
+        } else if dependencies.recorder.isInternal {
             dependencies.recorder.stop()
```

Comment:
Here we should call `stop(force: true)`. This bug ins't introduced here and it even shouldn't affect current users since all the telemetry is removed when `China` is detected.
I noticed that when realized that `SessionRecorderProtocol` hides `force` option of `SessionRecorder` method.
This behavior will be adjusted in https://github.com/mapbox/mapbox-vision/issues/1312 so there's no need to change anything. Just putting it here for the record.